Marlin/src/module/configuration_store.cpp

```diff
@@ -2281,6 +2334,24 @@ void MarlinSettings::reset() {
     #endif
   #endif
 
+  //
+  // Baud Rate
+  //
+  {
+    #if ENABLED(BAUD_RATE_GCODE)
+      #ifdef SERIAL_PORT
+        if (MYSERIAL0.baudrate != BAUDRATE) {
+          MYSERIAL0.end(); MYSERIAL0.begin(BAUDRATE);
+        }
+      #endif
+      #ifdef SERIAL_PORT_2
+        if (MYSERIAL1.baudrate != BAUDRATE) {
+          MYSERIAL1.end(); MYSERIAL1.begin(BAUDRATE);
+        }
+      #endif
+    #endif
+  }
+
```

Comment:
Predicted support flood: "Why does M502 disconnect me from the host?"

Reply:
If hosts dont implement the notify we just talked about, I anticipate a flood of those in alot of places, including during bootup as things sit as it detects the hard coded baud then switches...

Reply:
It's a very niche feature. Driving it into the mainstream with more functionality and visibility is good and completist, but is it a benefit to the universe as a whole?

Reply:
As it was it had almost no use as it only built for the 2 platforms that supported the overload class in Marlin (due and avr) and would trigger a reconnect on AVR for sure immediately losing the change. I dont know if DUE reboots on connect, but if so the issue would be there as well. Some technicians terminals may be able to switch on the fly, but hosts will all time out then reconnect. Whats been added here was basically the minimum to make it useful in its current state. I agree that its niche, and probably only useful in high noise environments where a host could in theory track resend requests and if they exceed a threshold request a change before continuing. On the other hand its a step towards parity with RRF and such.

Reply:
Yes, the feature is largely pointless without also saving the baud rate so that reconnect works. But note that changing the baud rate in the middle of EEPROM operations is troublesome. It seems to me that the saved baud rate needs to be applied at the standard serial initialization time, and that means the serial initialization must take place after loading EEPROM. However, we use serial reporting in EEPROM loading. So… it is a typical can of worms.

Reply:
And that's exactly why on the note about reorganizing the setup routine I tagged you! That is a tough one to decide the best course of action. I dropped in something that will at least work fully aware that it will need a good bit of optimization and adjustment to be right.

Reply:
One thought that I had by the way as a potential solution is I believe we have one or two other features using the first 100 bites with a fixed address for their eeprom storage outside the normal structure. We could put in individual address read in the front section to initialize the baud rate prior to the generic configuration load.

Reply:
> we have one or two other features using the first 100 bytes

We shouldn't. Last time I tried setting that value to 0 instead of 100 and it caused the EEPROM to fail completely — wouldn't read or write. I thought perhaps that means the offset of 100 is important.

If the baud rate can be stored as the very first EEPROM value, that will also work fine. But note that serial reporting is happening before the EEPROM loads also. Some of that pre-initialization may be important.